homeassistant/components/template/sensor.py

```diff
@@ -291,3 +301,15 @@ def should_poll(self):
                         self._name,
                         ex,
                     )
+        if self._availability_template is not None:
```

Comment:
Use the mechanics from https://github.com/home-assistant/home-assistant/blob/8fb273f00c3b0a963d9e636d659665b81a96b265/homeassistant/components/template/sensor.py#L261

Reply:
hey @pvizeli ,
 I understand what you mean, but the result of the render for 'availability_template' is processed differently to the `icon_template`,`entity_picture_template` and `friendly_name_template` templates. 
For these 3 templates the result is stored verbatim in the respective instance vars: 
https://github.com/home-assistant/home-assistant/blob/8fb273f00c3b0a963d9e636d659665b81a96b265/homeassistant/components/template/sensor.py#L281
whereas for the `availability_template` the result is compared to `true` to set the instance variable `_availabile` to a boolean:
https://github.com/home-assistant/home-assistant/blob/8fb273f00c3b0a963d9e636d659665b81a96b265/homeassistant/components/template/sensor.py#L306

I am working on a refactor for the template intialisation and rendering for all the template platform components to remove  a bunch of duplication, but want to get these pushed through first as the refactor creates change in all the components.

Does that make sense?

Reply:
Thinking about it I could make the render for the `availability` template store the result in the `_availabile` as the direct result of the render and then change the `def available():` method to do the comparison to `true`.

Sure it's a bit more processing whenever `available()` is called, but it would remove the duplication.

Would that work for you?

Reply:
Fixed it as mentioned above.

Reply:
Also fixed it where nescessary on the 7 other template platforms